### PR TITLE
Add description on ALTER DATABASE SET TIFLASH REPLICA

### PR DESCRIPTION
**Pull request by CalvinNeo**
Signed-off-by: CalvinNeo <calvinneo1995@gmail.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiflash/use-tiflash.md [hunk 1]
```diff
@@ -151,6 +151,61 @@ SELECT * FROM information_schema.tiflash_replica WHERE TABLE_SCHEMA = '<db_name>
...
+
+删除副本：
+
+{{< copyable "sql" >}}
+
+```sql
+ALTER TABLE `tpch50` SET TIFLASH REPLICA 0
+```
+
+注意事项：
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:184]:
```suggestion
注意事项：

```

File: tiflash/use-tiflash.md [hunk 2]
```diff
@@ -151,6 +151,61 @@ SELECT * FROM information_schema.tiflash_replica WHERE TABLE_SCHEMA = '<db_name>
...
+1. 该操作实际是为用户执行一系列 DDL 操作。
+1. 从命令执行开始到该库中所有表都已**同步完成**之前，若对该库下的表再执行 DDL ，则最终状态可能非预期。包括：
+    设置 TIFLASH REPLICA 数量分别为 2，在库中所有的表都同步完成前，再设置 TIFLASH REPLICA 数量为 1，不能保证最终所有表的 replica 数量都为 1 或都为 2。
+    从命令执行开始到执行结束，如果期间在该库下创建表，则**可能**会对这些新增表创建 TIFLASH REPLICA。
+    从命令执行开始到执行结束，如果期间对该库下的表添加索引，则该命令可能陷入等待，直到添加索引完成。
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:193]:
```suggestion
+ 该操作实际是为用户执行一系列 DDL 操作。
+ 从命令执行开始到该库中所有表都已**同步完成**之前，若对该库下的表再执行 DDL ，则最终状态可能非预期。包括：
    + 设置 TIFLASH REPLICA 数量分别为 2，在库中所有的表都同步完成前，再设置 TIFLASH REPLICA 数量为 1，不能保证最终所有表的 replica 数量都为 1 或都为 2。
    + 从命令执行开始到执行结束，如果期间在该库下创建表，则**可能**会对这些新增表创建 TIFLASH REPLICA。
    + 从命令执行开始到执行结束，如果期间对该库下的表添加索引，则该命令可能陷入等待，直到添加索引完成。
```

File: tiflash/use-tiflash.md [hunk 3]
```diff
@@ -151,6 +151,61 @@ SELECT * FROM information_schema.tiflash_replica WHERE TABLE_SCHEMA = '<db_name>
...
+
+
+### 查看库同步进度
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:196]:
```suggestion
### 查看库同步进度

```

File: tiflash/use-tiflash.md [hunk 4]
```diff
@@ -151,14 +151,74 @@ SELECT * FROM information_schema.tiflash_replica WHERE TABLE_SCHEMA = '<db_name>
...
+
+{{< copyable "sql" >}}
+
+```sql
+ALTER DATABASE `tpch50` SET TIFLASH REPLICA 0
+```
+
+注意事项：
+
++ 该命令实际是为用户执行一系列 DDL 操作。如果在执行过程中出现中断，未执行的操作不会继续执行。
```
Comment (LiSong0214) [inline tiflash/use-tiflash.md:186]: 已经执行的操作不会回退

File: tiflash/use-tiflash.md [hunk 5]
```diff
@@ -151,14 +151,74 @@ SELECT * FROM information_schema.tiflash_replica WHERE TABLE_SCHEMA = '<db_name>
...
+    
++ 从命令执行开始到该库中所有表都已**同步完成**之前，若对该库下的表再执行 DDL ，则最终状态可能非预期。包括：
```
Comment (LiSong0214) [inline tiflash/use-tiflash.md]: 因此，不建议用户在此期间执行其他 DDL 操作

**Review by LiSong0214 — APPROVED** [review]

File: tiflash/use-tiflash.md [hunk 6]
```diff
@@ -151,6 +151,68 @@ SELECT * FROM information_schema.tiflash_replica WHERE TABLE_SCHEMA = '<db_name>
...
+
+```sql
+ALTER DATABASE `tpch50` SET TIFLASH REPLICA 0
+```
+
+注意事项：
+
++ 该命令实际是为用户执行一系列 DDL 操作，对系统资源要求较高。如果在执行过程中出现中断，已经执行成功的操作不会回退，未执行的操作不会继续执行。
+    
++ 从命令执行开始到该库中所有表都已**同步完成**之前，不建议执行和该库相关的设置 TiFlash 副本或其他 DDL 操作，否则最终状态可能非预期。包括：
```
Comment (shichun-0415) [inline tiflash/use-tiflash.md:188]:
```suggestion
+ 从命令执行开始到该库中所有表都已**同步完成**之前，不建议执行和该库相关的设置 TiFlash 副本或其他 DDL 操作，否则最终状态可能非预期。非预期场景包括：
```

File: tiflash/use-tiflash.md [hunk 7]
```diff
@@ -151,6 +151,68 @@ SELECT * FROM information_schema.tiflash_replica WHERE TABLE_SCHEMA = '<db_name>
...
+
+    + 先设置 TiFlash 副本数量为 2，在库中所有的表都同步完成前，再设置 TIFLASH 副本数量为 1，不能保证最终所有表的 TiFlash 副本数量都为 1 或都为 2。
+
+    + 从命令执行开始到执行结束，如果期间在该库下创建表，则**可能**会对这些新增表创建 TiFlash 副本。
+    
+    + 从命令执行开始到执行结束，如果期间对该库下的表添加索引，则该命令可能陷入等待，直到添加索引完成。
+
++ 该命令会跳过系统表、视图、临时表，以及使用 TiFlash 不支持字符集的表。
```
Comment (shichun-0415) [inline tiflash/use-tiflash.md:196]:
```suggestion
    + 先设置 TiFlash 副本数量为 2，在库中所有的表都同步完成前，再设置 TiFlash 副本数量为 1，不能保证最终所有表的 TiFlash 副本数量都为 1 或都为 2。

    + 在命令执行到结束期间，如果在该库下创建表，则**可能**会对这些新增表创建 TiFlash 副本。
    
    + 在命令执行到结束期间，如果为该库下的表添加索引，则该命令可能陷入等待，直到添加索引完成。

+ 该命令会跳过系统表、视图、临时表以及使用 TiFlash 不支持字符集的表。
```

File: tiflash/use-tiflash.md [hunk 8]
```diff
@@ -151,6 +151,68 @@ SELECT * FROM information_schema.tiflash_replica WHERE TABLE_SCHEMA = '<db_name>
...
+
+### 查看库同步进度
+
+该命令在设置完副本后即返回，而不会等待所有表都同步完成。可以通过下面的 SQL 检查数据库中所有已设置 TiFlash Replica 表的同步进度
```
Comment (shichun-0415) [inline tiflash/use-tiflash.md:200]: “该命令”指哪个命令？跟下面的 SQL 啥关系？

Comment (shichun-0415) [inline tiflash/use-tiflash.md:200]:
```suggestion
该命令在设置完副本后即返回，而不会等待所有表都同步完成。可以通过下面的 SQL 检查数据库中所有已设置 TiFlash Replica 表的同步进度：
```

File: tiflash/use-tiflash.md [hunk 9]
```diff
@@ -151,6 +151,68 @@ SELECT * FROM information_schema.tiflash_replica WHERE TABLE_SCHEMA = '<db_name>
...
+
+{{< copyable "sql" >}}
+
+```sql
+SELECT * FROM information_schema.tiflash_replica WHERE TABLE_SCHEMA = '<db_name>'
+```
+
+可以通过下面的 SQL 检查数据库中尚未设置 TiFlash Replica 的表名
```
Comment (shichun-0415) [inline tiflash/use-tiflash.md:208]:
```suggestion
可以通过下面的 SQL 检查数据库中尚未设置 TiFlash Replica 的表名：
```

File: tiflash/use-tiflash.md [hunk 10]
```diff
@@ -151,6 +151,68 @@ SELECT * FROM information_schema.tiflash_replica WHERE TABLE_SCHEMA = '<db_name>
...
+
+    + 在命令执行到结束期间，如果在该库下创建表，则**可能**会对这些新增表创建 TiFlash 副本。
+    
+    + 在命令执行到结束期间，如果为该库下的表添加索引，则该命令可能陷入等待，直到添加索引完成。
+
++ 该命令会跳过系统表、视图、临时表以及使用 TiFlash 不支持字符集的表。
+
+### 查看库同步进度
+
+类似于按表构建，按库构建 TiFlash 副本的命令执行成功同样不代表所有表都已同步完成。可以通过下面的 SQL 检查数据库中所有已设置 TiFlash Replica 表的同步进度：
```
Comment (shichun-0415) [inline tiflash/use-tiflash.md:200]:
```suggestion
类似于按表构建，按库构建 TiFlash 副本的命令执行成功，不代表所有表都已同步完成。可以执行下面的 SQL 语句检查数据库中所有已设置 TiFlash Replica 表的同步进度：
```

File: tiflash/use-tiflash.md [hunk 11]
```diff
@@ -151,6 +151,68 @@ SELECT * FROM information_schema.tiflash_replica WHERE TABLE_SCHEMA = '<db_name>
...
+
+{{< copyable "sql" >}}
+
+```sql
+SELECT * FROM information_schema.tiflash_replica WHERE TABLE_SCHEMA = '<db_name>'
+```
+
+可以通过下面的 SQL 检查数据库中尚未设置 TiFlash Replica 的表名：
```
Comment (shichun-0415) [inline tiflash/use-tiflash.md:208]:
```suggestion
可以执行下面的 SQL 语句检查数据库中尚未设置 TiFlash Replica 的表名：
```

**Review by shichun-0415 — APPROVED** [review]
LGTM

File: tiflash/use-tiflash.md [hunk 12]
```diff
@@ -151,6 +151,68 @@ SELECT * FROM information_schema.tiflash_replica WHERE TABLE_SCHEMA = '<db_name>
 
 关于使用 label 进行副本调度划分可用区的更多内容，可以参考[通过拓扑 label 进行副本调度](/schedule-replicas-by-topology-labels.md)，[同城多数据中心部署 TiDB](/multi-data-centers-in-one-city-deployment.md) 与[两地三中心部署](/three-data-centers-in-two-cities-deployment.md)。
 
+## 按库构建 TiFlash 副本
+
+类似于按表构建 TiFlash 副本的方式，可通过 MySQL 客户端向 TiDB 发送 DDL 命令来为特定的库中的所有表建立 TiFlash 副本：
```
Comment (qiancai) [inline tiflash/use-tiflash.md:156]:
```suggestion
类似于按表构建 TiFlash 副本的方式，你可以在 MySQL 客户端向 TiDB 发送 DDL 命令来为指定数据库中的所有表建立 TiFlash 副本：
```

File: tiflash/use-tiflash.md [hunk 13]
```diff
@@ -151,6 +151,68 @@ SELECT * FROM information_schema.tiflash_replica WHERE TABLE_SCHEMA = '<db_name>
...
+
+{{< copyable "sql" >}}
+
+```sql
+ALTER DATABASE db_name SET TIFLASH REPLICA count
+```
+
+该命令的参数说明如下：
+
+- count 表示副本数，0 表示删除。
```
Comment (qiancai) [inline tiflash/use-tiflash.md:166]:
```suggestion
在该命令中，`${count}` 表示 TiFlash 的副本数。当设置 `${count}` 值为 0 时，表示删除现有的 TiFlash 副本。
````

File: tiflash/use-tiflash.md [hunk 14]
```diff
@@ -151,6 +151,68 @@ SELECT * FROM information_schema.tiflash_replica WHERE TABLE_SCHEMA = '<db_name>
...
 关于使用 label 进行副本调度划分可用区的更多内容，可以参考[通过拓扑 label 进行副本调度](/schedule-replicas-by-topology-labels.md)，[同城多数据中心部署 TiDB](/multi-data-centers-in-one-city-deployment.md) 与[两地三中心部署](/three-data-centers-in-two-cities-deployment.md)。
 
+## 按库构建 TiFlash 副本
+
+类似于按表构建 TiFlash 副本的方式，可通过 MySQL 客户端向 TiDB 发送 DDL 命令来为特定的库中的所有表建立 TiFlash 副本：
+
+{{< copyable "sql" >}}
+
+```sql
+ALTER DATABASE db_name SET TIFLASH REPLICA count
```
Comment (qiancai) [inline tiflash/use-tiflash.md:161]:
```suggestion
ALTER DATABASE db_name SET TIFLASH REPLICA ${count}
```

File: tiflash/use-tiflash.md [hunk 15]
```diff
@@ -151,6 +151,68 @@ SELECT * FROM information_schema.tiflash_replica WHERE TABLE_SCHEMA = '<db_name>
...
+```
+
+该命令的参数说明如下：
+
+- count 表示副本数，0 表示删除。
+
+为库中的所有表建立 2 个副本：
```
Comment (qiancai) [inline tiflash/use-tiflash.md:168]:
```suggestion
命令示例：

执行以下命令可以为 `tpch50` 库中的所有表建立 2 个 TiFlash 副本。
```

File: tiflash/use-tiflash.md [hunk 16]
```diff
@@ -151,6 +151,68 @@ SELECT * FROM information_schema.tiflash_replica WHERE TABLE_SCHEMA = '<db_name>
...
+
+{{< copyable "sql" >}}
+
+```sql
+ALTER DATABASE `tpch50` SET TIFLASH REPLICA 2
+```
+
+删除副本：
```
Comment (qiancai) [inline tiflash/use-tiflash.md:176]:
```suggestion
执行以下命令可以删除为 `tpch50` 库建立的 TiFlash 副本：
```

File: tiflash/use-tiflash.md [hunk 17]
```diff
@@ -151,6 +151,68 @@ SELECT * FROM information_schema.tiflash_replica WHERE TABLE_SCHEMA = '<db_name>
...
+
+```sql
+ALTER DATABASE `tpch50` SET TIFLASH REPLICA 0
+```
+
+注意事项：
+
++ 该命令实际是为用户执行一系列 DDL 操作，对资源要求比较高。如果在执行过程中出现中断，已经执行成功的操作不会回退，未执行的操作不会继续执行。
+    
++ 从命令执行开始到该库中所有表都已**同步完成**之前，不建议执行和该库相关的设置 TiFlash 副本或其他 DDL 操作，否则最终状态可能非预期。非预期场景包括：
```
Comment (qiancai) [inline tiflash/use-tiflash.md:188]:
```suggestion
+ 从命令执行开始到该库中所有表都已**同步完成**之前，不建议执行和该库相关的 TiFlash 副本数量设置或其他 DDL 操作，否则最终状态可能非预期。非预期场景包括：
```

File: tiflash/use-tiflash.md [hunk 18]
```diff
@@ -151,6 +151,68 @@ SELECT * FROM information_schema.tiflash_replica WHERE TABLE_SCHEMA = '<db_name>
...
+
+    + 先设置 TiFlash 副本数量为 2，在库中所有的表都同步完成前，再设置 TiFlash 副本数量为 1，不能保证最终所有表的 TiFlash 副本数量都为 1 或都为 2。
+
+    + 在命令执行到结束期间，如果在该库下创建表，则**可能**会对这些新增表创建 TiFlash 副本。
+    
+    + 在命令执行到结束期间，如果为该库下的表添加索引，则该命令可能陷入等待，直到添加索引完成。
+
++ 该命令会跳过系统表、视图、临时表以及使用 TiFlash 不支持字符集的表。
```
Comment (qiancai) [inline tiflash/use-tiflash.md:196]:
```suggestion
+ 该命令会跳过系统表、视图、临时表以及包含了 TiFlash 不支持字符集的表。
```

File: tiflash/use-tiflash.md (hunk 14 above)
Comment (CalvinNeo) [reply tiflash/use-tiflash.md:161]: db_name 是否也要这么处理呢？“按表构建 TiFlash 副本”这个既有章节里面都没有加

Comment (qiancai) [reply tiflash/use-tiflash.md:161]: 那还是不改了吧😂同一篇文档先自己内部统一。

File: tiflash/use-tiflash.md [hunk 19]
```diff
@@ -151,6 +151,68 @@ SELECT * FROM information_schema.tiflash_replica WHERE TABLE_SCHEMA = '<db_name>
...
+
+类似于按表构建 TiFlash 副本的方式，你可以在 MySQL 客户端向 TiDB 发送 DDL 命令来为指定数据库中的所有表建立 TiFlash 副本：
+
+{{< copyable "sql" >}}
+
+```sql
+ALTER DATABASE db_name SET TIFLASH REPLICA count
+```
+
+在该命令中，`${count}` 表示 TiFlash 的副本数。当设置 `${count}` 值为 0 时，表示删除现有的 TiFlash 副本。
```
Comment (qiancai) [inline tiflash/use-tiflash.md:164]:
```suggestion
在该命令中，`count` 表示 TiFlash 的副本数。当设置 `count` 值为 0 时，表示删除现有的 TiFlash 副本。
```

File: tiflash/use-tiflash.md (hunk 14 above)
Comment (CalvinNeo) [reply tiflash/use-tiflash.md:161]: OK

**Review by qiancai — APPROVED** [review]
LGTM

File: tiflash/use-tiflash.md [hunk 20]
```diff
@@ -151,6 +151,65 @@ SELECT * FROM information_schema.tiflash_replica WHERE TABLE_SCHEMA = '<db_name>
...
+> **注意：**
+>
+> - 该命令实际是为用户执行一系列 DDL 操作，对资源要求比较高。如果在执行过程中出现中断，已经执行成功的操作不会回退，未执行的操作不会继续执行。
+>
+> - 从命令执行开始到该库中所有表都已**同步完成**之前，不建议执行和该库相关的 TiFlash 副本数量设置或其他 DDL 操作，否则最终状态可能非预期。非预期场景包括：
+>     - 先设置 TiFlash 副本数量为 2，在库中所有的表都同步完成前，再设置 TiFlash 副本数量为 1，不能保证最终所有表的 TiFlash 副本数量都为 1 或都为 2。
+>     - 在命令执行到结束期间，如果在该库下创建表，则**可能**会对这些新增表创建 TiFlash 副本。
+>     - 在命令执行到结束期间，如果为该库下的表添加索引，则该命令可能陷入等待，直到添加索引完成。
+
++ 该命令会跳过系统表、视图、临时表以及包含了 TiFlash 不支持字符集的表。
```
Comment (shichun-0415) [inline tiflash/use-tiflash.md:193]:
```suggestion
>     - 在命令执行到结束期间，如果为该库下的表添加索引，则该命令可能陷入等待，直到添加索引完成。
>
> - 该命令会跳过系统表、视图、临时表以及包含了 TiFlash 不支持字符集的表。
```

**Pull request merged**